### PR TITLE
Fix video embeddings + Add asciinema

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Frontend

### Description
This PR:
- fixes the ratio calculation of vimeo and youtube embedding
- readds asciinema embeddings

### Steps

- [x] Added implementation
- [x] Added / updated tests
- [x] Added changelog snippet
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc/blob/develop/CONTRIBUTING.md) and
  made sure that:
  - My commits are signed-off to accept the DCO.
  - This PR targets the correct branch: `master` for 1.x & docs, `develop` for 2.x

### CODE REVIEW

File: frontend/src/extensions/extra-integrations/asciinema/asciinema-app-extension.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import type { MarkdownRendererExtension } from '../../../components/markdown-renderer/extensions/base/markdown-renderer-extension'
+import { AppExtension } from '../../base/app-extension'
+import { AsciinemaMarkdownExtension } from './asciinema-markdown-extension'
+
+/**
+ * Adds support for LaTeX rendering using KaTeX to the markdown rendering.
+ *
+ * @see https://katex.org/
```
Comment (DerMolly) [inline frontend/src/extensions/extra-integrations/asciinema/asciinema-app-extension.ts:13]: Needs to be updated

File: frontend/src/extensions/extra-integrations/asciinema/asciinema-frame.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import React from 'react'
+
+/**
+ * Renders an embedding for https://asciinema.org
+ *
+ * @param id The id from the asciinema url
+ */
+export const AsciinemaFrame: React.FC<IdProps> = ({ id }) => {
+  return (
+    <ClickShield
```
Comment (DerMolly) [inline frontend/src/extensions/extra-integrations/asciinema/asciinema-frame.tsx:17]: Shouldn't we add a key property with the asciinema id here?

File: frontend/src/extensions/extra-integrations/asciinema/asciinema-markdown-extension.test.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import React from 'react'
+
+jest.mock('./asciinema-frame')
+
+describe('asciinema markdown extension', () => {
+  beforeAll(async () => {
+    jest
+      .spyOn(AsciinemaFrameModule, 'AsciinemaFrame')
+      .mockImplementation((({ id }) => (
+        <span>this is a mock for the youtube frame with id {id}</span>
```
Comment (DerMolly) [inline frontend/src/extensions/extra-integrations/asciinema/asciinema-markdown-extension.test.tsx:21]:
```suggestion
        <span>this is a mock for the asciinema frame with id {id}</span>
```

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: frontend/src/extensions/extra-integrations/asciinema/asciinema-app-extension.ts (hunk 1 above)
Comment (mrdrogdrog) [reply frontend/src/extensions/extra-integrations/asciinema/asciinema-app-extension.ts:13]: why?

File: frontend/src/extensions/extra-integrations/asciinema/asciinema-frame.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply frontend/src/extensions/extra-integrations/asciinema/asciinema-frame.tsx:17]: Not necessary because ids are only valid under one parent.

**Review by DerMolly — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**